scripts/tools/zap_regen_all.py

```diff
@@ -65,6 +66,50 @@ class TargetType(Flag):
 }
 
 
+class ZapInput:
+    """ZAP may be run from a .zap configuration or from zcl properties.
+
+    Running from a '.zap' configuration will load the zcl properties from
+    the '.zap' file, however it will also load cluster enabling and settings.
+
+    For `client-side` code generation, CHIP wants to explicitly not depend
+    on zap enabling/disabling as everything should be enabled.
+    """
+
+    @staticmethod
+    def FromZap(f):
+        return ZapInput(zap_file=str(f))
+
+    @staticmethod
+    def FromZcl(f):
```

Comment:
I think it's worth not calling the XML "zcl" in any of this stuff, even if ZAP itself calls the option `-z`.

Reply:
I went by zap, where the option is:

```
-z, --zclProperties, --zcl          zcl.properties file to read in...
```

Unsure what else to call itl, everything is "zcl" and I wish I knew what the acronym stands for (zap ... configuration list?).

Reply:
Updated, however this is still not XML (the argument given that is) so unsure if the rename is worth it.

We can do any additional renames as a followup. I would have named it something `XML` if I did not have to pass in the zcl file ... however currently it seems I have to pass it in.

Reply:
"zcl" is "Zigbee Cluster Library" @andy31415